File: 06600/BOJ_6603.py
import sys

input = sys.stdin.readline

def backtracking(idx):
    if len(arr) == 6:
        print(*arr)
        return
    for i in range(idx, k):
        arr.append(s[i])
        backtracking(i+1)
        arr.pop()

while True:
    lst = list(map(int, input().split()))
    k = lst[0]
    s = lst[1:]
    if k == 0:
        sys.exit(0)
    arr = []
    backtracking(0)
    print()

File: 06600/BOJ_6603.swift
import Foundation

func backtracking(s: [Int], arr: inout [Int], k: Int, idx: Int) {
    if arr.count == 6 {
        print(arr.map { String($0) }.joined(separator: " "))
        return
    }
    
    for i in stride(from: idx, to: k, by: 1) {
        arr.append(s[i])
        backtracking(s: s, arr: &arr, k: k, idx: i+1)
        arr.removeLast()
    }
}

while true {
    let input = readLine()!.split(separator: " ").map {Int($0)!}
    let k = input[0]
    let s = Array(input[1...])
    if k == 0 {
        exit(0)
    }
    var arr: [Int] = []
    backtracking(s: s, arr: &arr, k: k, idx: 0)
    print()
}
